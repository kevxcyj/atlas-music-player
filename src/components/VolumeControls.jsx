import { Volume2 } from 'lucide-react';

const VolumeControls = () => {
  return (
    <div className="flex items-center space-x-2 w-full">
      <button>
        <Volume2 size={24} className="text-black dark:text-white" />
      </button>
      <input
        type="range"
        className="h-2 w-full cursor-pointer appearance-none rounded-lg bg-gray-200"
        min="0"
        max="100"
      />
    </div>
  );
};

export default VolumeControls;