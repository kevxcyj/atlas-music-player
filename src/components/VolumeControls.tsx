import { Volume2 } from 'lucide-react';

interface VolumeControlsProps {
  volume: number;
  onVolumeChange: (newVolume: number) => void;
}

const VolumeControls: React.FC<VolumeControlsProps> = ({ volume, onVolumeChange }) => {
  return (
    <div className="flex items-center space-x-2 w-full">
      <button aria-label="Volume Icon">
        <Volume2 size={24} className="text-black dark:text-white" />
      </button>
      <input
        type="range"
        aria-label="Volume Slider"
        className="h-2 w-full cursor-pointer appearance-none rounded-lg bg-gray-200"
        min="0"
        max="100"
        value={volume}
        onChange={(e) => onVolumeChange(Number(e.target.value))}
      />
    </div>
  );
};

export default VolumeControls;