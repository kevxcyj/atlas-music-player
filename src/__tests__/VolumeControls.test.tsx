import { render } from '@testing-library/react';
import { expect, test, vi } from 'vitest';
import VolumeControls from '../components/VolumeControls';

test('VolumeControls renders correctly with default props', () => {
  const { container } = render(<VolumeControls volume={50} onVolumeChange={() => {}} />);
  expect(container).toMatchSnapshot();
});

test('VolumeControls renders correctly with a different volume', () => {
  const { container } = render(<VolumeControls volume={25} onVolumeChange={() => {}} />);
  expect(container).toMatchSnapshot();
});