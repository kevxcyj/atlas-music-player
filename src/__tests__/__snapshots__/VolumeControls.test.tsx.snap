// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls renders correctly with a different volume 1`] = `
<div>
  <div
    class="flex items-center space-x-2 w-full"
  >
    <button
      aria-label="Volume Icon"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-volume2 lucide-volume-2 text-black dark:text-white"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
        />
        <path
          d="M16 9a5 5 0 0 1 0 6"
        />
        <path
          d="M19.364 18.364a9 9 0 0 0 0-12.728"
        />
      </svg>
    </button>
    <input
      aria-label="Volume Slider"
      class="h-2 w-full cursor-pointer appearance-none rounded-lg bg-gray-200"
      max="100"
      min="0"
      type="range"
      value="25"
    />
  </div>
</div>
`;

exports[`VolumeControls renders correctly with default props 1`] = `
<div>
  <div
    class="flex items-center space-x-2 w-full"
  >
    <button
      aria-label="Volume Icon"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-volume2 lucide-volume-2 text-black dark:text-white"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
        />
        <path
          d="M16 9a5 5 0 0 1 0 6"
        />
        <path
          d="M19.364 18.364a9 9 0 0 0 0-12.728"
        />
      </svg>
    </button>
    <input
      aria-label="Volume Slider"
      class="h-2 w-full cursor-pointer appearance-none rounded-lg bg-gray-200"
      max="100"
      min="0"
      type="range"
      value="50"
    />
  </div>
</div>
`;
